python/xorbits/experimental/dedup.py

```diff
@@ -478,55 +246,78 @@ def dedup(
     if col not in df.dtypes.index:
         raise ValueError(f"{col} column not found in the DataFrame")
 
-    B, R = optimal_param(threshold, num_perm)
+    if method == "exact":
+        df = to_mars(df)
+
+        df["__exact"] = df[col].apply(
+            lambda x: hashlib.md5(x.encode("utf-8")).hexdigest()
+        )
+
+        df.drop_duplicates(subset=["__exact"], inplace=True).drop(
+            columns=["__exact"], inplace=True
+        )
 
-    HASH_RANGES = [(i * R, (i + 1) * R) for i in range(B)]
+        return from_mars(df)
 
-    RNG = np.random.RandomState(seed)
+    if method == "minhash":
+        threshold = kws.pop("threshold", 0.7)
+        num_perm = kws.pop("num_perm", 128)
+        min_length = kws.pop("min_length", 5)
+        ngrams = kws.pop("ngrams", 5)
+        seed = kws.pop("seed", 42)
 
-    PERMUTATIONS = np.array(
-        [
-            (
-                RNG.randint(1, MERSENNE_PRIME, dtype=np.uint64),
-                RNG.randint(0, MERSENNE_PRIME, dtype=np.uint64),
+        # Check the threshold type and range
+        if not isinstance(threshold, (float, int)) or not 0 <= threshold <= 1:
+            raise ValueError(
+                f"Expected 'threshold' to be a float between 0 and 1, got {threshold}"
             )
-            for _ in range(num_perm)
-        ],
-        dtype=np.uint64,
-    ).T
-
-    func = partial(
-        embed_func,
-        text=col,
-        num_perm=num_perm,
-        hashranges=HASH_RANGES,
-        ngram_size=ngram,
-        min_length=min_length,
-        permutations=PERMUTATIONS,
-    )
-
-    op = DataFrameDedup(func=func)
-
-    return from_mars(op(to_mars(df)))
+
+        # Check the num_perm, min_length, ngram and seed type and value
+        for var, var_name in [
+            (num_perm, "num_perm"),
+            (min_length, "min_length"),
+            (ngrams, "ngrams"),
+            (seed, "seed"),
+        ]:
+            if not isinstance(var, int) or var <= 0:
+                raise ValueError(
+                    f"Expected '{var_name}' to be a positive integer, got {var}"
+                )
+
+        B, R = optimal_param(threshold, num_perm)
+
+        HASH_RANGES = [(i * R, (i + 1) * R) for i in range(B)]
+
+        RNG = np.random.RandomState(seed)
+
+        PERMUTATIONS = np.array(
+            [
+                (
+                    RNG.randint(1, MERSENNE_PRIME, dtype=np.uint64),
+                    RNG.randint(0, MERSENNE_PRIME, dtype=np.uint64),
+                )
+                for _ in range(num_perm)
+            ],
+            dtype=np.uint64,
+        ).T
+
+        func = partial(
+            minhash_embed_func,
+            text=col,
+            num_perm=num_perm,
+            hashranges=HASH_RANGES,
+            ngram_size=ngrams,
+            min_length=min_length,
+            permutations=PERMUTATIONS,
+        )
+
+        op = DataFrameDedup(func=func)
+
+        return from_mars(op(to_mars(df)))
```

Comment:
Raise NotImplementedError for not supported methods.

Reply:
I check the method in https://github.com/xorbitsai/xorbits/blob/4c6b572e0c4fec65190f14be3fa13bbcd8223a95/python/xorbits/experimental/dedup.py#L258